### PR TITLE
Convert  `pkg/util/winutil`  to module

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?
PR converts `pkg/util/winutil` to go module.  In the process we do the following 
- Move `pkg/util/winutil/crashreport` to `pkg/util/crashreport` - to simplify creating `pkg/util/winutil` module
- Move `pkg/util/winutil/pdhutil` to `pkg/util/pdhutil`  - to simplify creating `pkg/util/winutil` module

### Describe how to test/QA your changes

- create a windows vm
- install IIS
- powershell: Install-WindowsFeature -name Web-Server -IncludeManagementTools
- install the agent
- enable USM & NPM in system probe
- ensure you see your host in NPM
- after generating some http traffic, make sure you see defaultapppool in services for your host in USM
- 

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -351,6 +351,8 @@
 /pkg/util/cgroups/                      @DataDog/container-integrations
 /pkg/util/retry/                        @DataDog/container-integrations
 /pkg/util/intern/                       @DataDog/ebpf-platform
+/pkg/util/crashreport/                  @DataDog/windows-agent
```
Comment (iglendd) [inline .github/CODEOWNERS:354]: I think that crash report is created by @DataDog/windows-kernel-integrations (cc @derekwbrown)

**Review by iglendd — APPROVED** [review]
Perfect. Removing non-generic function from ETW is improves its implementation. Thank you. The only thing is setting ownership for crash reporting where I left the comment.

File: .github/CODEOWNERS (hunk 1 above)
Comment (derekwbrown) [reply .github/CODEOWNERS:354]: yes, that's correct

File: pkg/network/protocols/tls/java/testdata/.!10855!Wget$1.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/.!10855!Wget$1.class:1]: Please remove that file

**Review by guyarb — CHANGES_REQUESTED** [review]
Remove a file - https://github.com/DataDog/datadog-agent/pull/20851#discussion_r1393702635

File: pkg/network/protocols/tls/java/testdata/.!10855!Wget$1.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/.!10855!Wget$1.class:1]: remove

File: pkg/network/protocols/tls/java/testdata/TestAgentLoaded.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/TestAgentLoaded.class:1]: revert

File: pkg/network/protocols/tls/java/testdata/Wait.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/Wait.class:1]: revert

File: pkg/network/protocols/tls/java/testdata/Wget.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/Wget.class:1]: revert

File: pkg/network/protocols/tls/java/testdata/Wget$InvalidCertificateHostVerifier.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/Wget$InvalidCertificateHostVerifier.class:1]: revert

File: pkg/network/protocols/tls/java/testdata/Wget$2.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/Wget$2.class:1]: revert

File: pkg/network/protocols/tls/java/testdata/Wget$1.class
Comment (guyarb) [inline pkg/network/protocols/tls/java/testdata/Wget$1.class:1]: revert

**Review by guyarb — DISMISSED** [review]

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Review by kkhor-datadog — APPROVED** [review]

**Comment by dustmop** [comment]
Heads up that `secrets` is being turned into a component by https://github.com/DataDog/datadog-agent/pull/20927. This will impact this PR by both moving the source out of `pkg/secrets` (into `comp/core/secrets/secretsimpl`) and also by slightly changing the method signatures of secrets. Aside from that, it should not be changing any functionality.

File: pkg/secrets/go.mod [hunk 2]
```diff
@@ -0,0 +1,57 @@
+module github.com/DataDog/datadog-agent/pkg/secrets
```
Comment (hush-hush) [inline pkg/secrets/go.mod:1]: `pkg/secrets` was converted to a component `comp/core/secrets`

**Review by hush-hush — CHANGES_REQUESTED** [review]
This PR seems to do a lot more than converting secrets into a module. Could we either split the PR or update the PR's title ?

**Comment by dineshg13** [comment]
@hush-hush i have updated the PR title. 
@hush-hush @dustmop I will update the PR to convert `comp/core/secrets`  to a module.

**Review by hush-hush — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]

**Review by ISauve — APPROVED** [review]

**Comment by dineshg13** [comment]
We were able to verify it is working as expected 
image

**Pull request merged**